### PR TITLE
Add configuration for all Jackson Features

### PR DESCRIPTION
**Pull request by marcospereira**
## Purpose

Previously only [serialization](https://github.com/FasterXML/jackson-databind/wiki/Serialization-Features) and [deserialization](https://github.com/FasterXML/jackson-databind/wiki/Deserialization-Features) features were supported. This adds support for [MapperFeatures](https://github.com/FasterXML/jackson-databind/wiki/Mapper-Features), JsonParser and JsonGenerator features, making the `ObjectMapper` mostly configurable using `.conf` files.

## References

References #26870.

## Changes

- Support `MapperFeatures` configurations
- Support `JsonParser.Feature` configurations
- Support `JsonCreator.Feature` configurations

### CODE REVIEW

**Comment by jrudolph** [comment]
OK TO TEST

**Comment by jrudolph** [comment]
WHITELIST

File: akka-serialization-jackson/src/main/resources/reference.conf [hunk 1]
```diff
@@ -64,6 +64,21 @@ akka.serialization.jackson {
     FAIL_ON_UNKNOWN_PROPERTIES = off
   }
 
+  # Configuration of the ObjectMapper mapper features.
+  # See com.fasterxml.jackson.databind.MapperFeature
+  # Enum values corresponding to the MapperFeature and their boolean value.
```
Comment (jrudolph) [inline akka-serialization-jackson/src/main/resources/reference.conf]: > Enum values corresponding to the MapperFeature and their boolean value.

Seems a bit terse. I think we should also add examples to the comment.

Comment (marcospereira) [reply akka-serialization-jackson/src/main/resources/reference.conf]: Thanks, @jrudolph. See if https://github.com/akka/akka/pull/27409/commits/1b666fef3e1c9ea004729d9ac7c75a676e793545 addresses this comment. :-)

File: akka-serialization-jackson/src/main/resources/reference.conf [hunk 2]
```diff
@@ -64,6 +64,36 @@ akka.serialization.jackson {
...
+  # Configuration of the ObjectMapper JsonParser features.
+  # See com.fasterxml.jackson.core.JsonGenerator.Feature
+  # Enum values corresponding to the JsonGenerator.Feature and
+  # their boolean value, for example:
+  #
+  # json-generator-features {
+  #   WRITE_NUMBERS_AS_STRINGS = on
+  # }
+  json-generator-features {}
+
```
Comment (patriknw) [inline akka-serialization-jackson/src/main/resources/reference.conf]: Those will be included in the documentation, which is good.

**Review by patriknw — APPROVED** [review]
LGTM, thanks

File: akka-serialization-jackson/src/test/scala/akka/serialization/jackson/JacksonSerializerSpec.scala [hunk 3]
```diff
@@ -137,23 +140,88 @@ class JacksonJsonSerializerSpec extends JacksonSerializerSpec("jackson-json") {
...
+        val identifiedObjectMapper =
+          serialization(sys).serializerByIdentity(999).asInstanceOf[JacksonJsonSerializer].objectMapper
+        val namedObjectMapper = JacksonObjectMapperProvider(sys).getOrCreate("jackson-json2", None)
+        val defaultObjectMapper =
           serializerFor(ScalaTestMessages.SimpleCommand("abc")).asInstanceOf[JacksonJsonSerializer].objectMapper
-        objMapper.isEnabled(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES) should ===(false)
+
+        println(s"identifiedObjectMapper => $identifiedObjectMapper")
+        println(s"namedObjectMapper => $namedObjectMapper")
+        println(s"defaultObjectMapper => $defaultObjectMapper")
```
Comment (johanandren) [inline akka-serialization-jackson/src/test/scala/akka/serialization/jackson/JacksonSerializerSpec.scala]: Skip the printlns, use system.log if you really need output.

**Review by johanandren — COMMENTED** [review]
Not sure if I'm missing something but I can't find any test coverage for the three `overrideNnn` methods, would be good.

Looks good except for that.

**Comment by patriknw** [comment]
Fixed @johanandren 's review comments in 593a434

**Review by raboof — APPROVED** [review]

**Pull request merged**